RFC/src/RFC-0140 Syncing and seeding.md

```diff
@@ -0,0 +1,133 @@
+8. Validate blocks as if they where just mined and then received, in chronological order. 
+
+After this process the node will be in sync and able to process blocks and transaction normally. 
+
+#### Keeping in sync
+
+The node SHOULD periodically test its peers with ping messages to ensure that they are alive. When a node sends a ping message, it should include the current total PoW, hash of the [current head](currenthead) and genesis block hash of its own current longest chain in the ping message. The receiving node MUST reply with a pong message also including the total PoW, [current head](currenthead) and genesis block hash of its longest chain. If the two chains dont match up, the node with the lowest PoW has the responsiblity to ask the peer for syncing information. 
+
+If the genesis block hashes don't match, the node is removed from its peer list as this node is running a different blockchain. 
+
+This will be handled by the node asking for each block from the [current head](currenthead) going backward for older blocks until a known block is found. If no block is found, the node will enter sync mode and resync. It cannot recover from its state as the fork is older than its [pruning horizon](pruninghorizon).
+
+#### Chain forks
+
+Chain forks can be a problem since in MimbleWimble not all nodes keep the complete history, the design philosophy is more along the lines of only keeping the current [Blockchain state](blockchainstate). Only keeping the current [Blockchain state](blockchainstate) will not be possible, because if you encounter a fork where there are two running version on the blockchain, you will not be able to swop without doing sync again. 
+
+To counter this problem we use  [pruning horizon](pruninghorizon), this allows every [node](base node) to be a "light" [archival node](archivenode). This in effect means that the node will keep a full history for a short while. If the node encounters a fork it can easily rewind its state to apply the fork. If the fork is longer than the [pruning horizon](pruninghorizon), the node will enter a sync state where it will resync. 
+
+### Pruning and cut-through
+
+[Pruning and cut-through]: #Pruning-and-cut-through	"Remove already spent outputs from the [utxo]"
+
+In MimbleWimble, the state can be completely verified using the current [UTXO](utxo) set, the set of excess signatures (contained in the transaction kernels) and the proof-of-work. The full block and transaction history is not required. This allows base layer nodes to remove old used inputs from the [blockchain] and or the [mempool]. [Cut-through](cut-through) happens in the [mempool] while pruning happens in the [blockchain] with already confirmed transactions. This will remove the spent inputs and outputs, but will retain the excesses of each [transaction]. 
```

Comment:
```suggestion
In MimbleWimble, the state can be completely verified using the current [UTXO](utxo) set (which contains the spending commitments and range proofs), the set of excess signatures (contained in the transaction kernels) and the proof-of-work. The full block and transaction history is not required. This allows base layer nodes to remove old used inputs from the [blockchain] and or the [mempool]. [Cut-through](cut-through) happens in the [mempool] while pruning happens in the [blockchain] with already confirmed transactions. This will remove the spent inputs and outputs, but will retain the excesses of each [transaction]. 
```